BaikalDataGenerator/Source/data_generator.h

```diff
@@ -22,35 +22,59 @@ THE SOFTWARE.
 
 #pragma once
 
-#include "utils.h"
-#include "input_info.h"
-#include "filesystem.h"
+#include "Rpr/RadeonProRender.h"
 
-#include <vector>
+#ifdef __cplusplus
+extern "C" {
+#endif
 
-using CameraIterator = std::vector<CameraInfo>::const_iterator;
-using LightsIterator = std::vector<LightInfo>::const_iterator;
-using SppIterator = std::vector<int>::const_iterator;
-
-class ConfigLoader
+struct DataGeneratorParams
 {
-public:
-    explicit ConfigLoader(const DGenConfig& config);
+    char const* output_dir;
+
+    rpr_scene scene;
+    char const* scene_name;
+
+    rpr_camera* cameras;
+    unsigned cameras_start_idx;
+    unsigned cameras_num;
+    int cameras_offset_idx;
```

Comment:
Lets remove cameras_offset_idx from here and generate images for all passed cameras, starting output index from cameras_start_idx. Also lets split cameras outside datagenerator implementation - in the our application part